### PR TITLE
Add setting to choose between "selected", "all" or "random number" for dungeon shortcuts

### PR DESCRIPTION
**Pull request by RawZ06**
Adding a new setting to choice dungeon shortcuts
- Choice Dungeon is same as before
- All dungeon select all dungeon
- Random dungeon select a random number and dungeon with shortcuts

image

Note : If we select random, the list of dungeon with shortcuts is available in randomize_settings section in spoiler log.

### CODE REVIEW

File: SettingsList.py [hunk 1]
```diff
@@ -2430,22 +2430,16 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
-            'jabu_jabus_belly': 'Jabu Jabu\'s Belly',
-            'forest_temple': 'Forest Temple',
-            'fire_temple': 'Fire Temple',
-            'water_temple': 'Water Temple', # doesn't do anything, but added to prevent confusion
-            'shadow_temple': 'Shadow Temple',
-            'spirit_temple': 'Spirit Temple'
+            'off':       'Off',
+            'choice':    'Choice dungeons',
+            'all':	  'All dungeons',
+            'random': 'Random dungeons'
```
Comment (cjohnson57) [inline SettingsList.py:2441]: Nitpick: Spacing here is seemingly random, should align all of the options. Looks like the actual shortcut choices weren't aligned either so a good opportunity to do that as well.

File: tests/plando/boss_rush_dc_mq.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "shuffle_dungeon_entrances":               true,
   "open_forest":                             "open",
   "mq_dungeons":                             1,
-  "dungeon_shortcuts":                       ["dodongos_cavern"],
+  "dungeon_shortcuts":                       "choice",
+  "dungeon_shortcuts_list":                       ["dodongos_cavern"],
```
Comment (cjohnson57) [inline tests/plando/boss_rush_dc_mq.json:8]: These all should also be aligned.

**Review by cjohnson57 — CHANGES_REQUESTED** [review]
I'd rather not change the name of the existing setting when there's really no reason to, will just cause potential issues and confusion for no benefit. You could name your new setting something like `dungeon_shortcuts_choice` or `dungeon_shortcuts_setting`.

Edit: Actually, following the example of #1485 I think `dungeon_shortcuts_mode` would be the best.

File: SettingsList.py [hunk 3]
```diff
@@ -2467,6 +2461,39 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             <b>Spirit Temple</b>: lobby elevator activated,
             shortcut silver blocks moved, central room
             platform lowered, and statue face melted
+
+            Choice : Select dungeons with shortcuts
+            All : Enable all dungeons shortcuts
+            Random : Random dungeon shortcuts
```
Comment (cjohnson57) [inline SettingsList.py:2467]: Sorry forgot this on my first review, one last spacing nitpick: Should be no space directly before the colon.

File: SettingsList.py (hunk 1 above)
Comment (RawZ06) [reply SettingsList.py]: Fix : https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/b3958c886644d5d247a9adc47cb33b4e9a50bafb

File: tests/plando/boss_rush_dc_mq.json (hunk 2 above)
Comment (RawZ06) [reply tests/plando/boss_rush_dc_mq.json]: Fix : https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/b3958c886644d5d247a9adc47cb33b4e9a50bafb

File: SettingsList.py (hunk 3 above)
Comment (RawZ06) [reply SettingsList.py]: Fix https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/b3958c886644d5d247a9adc47cb33b4e9a50bafb

**Comment by RawZ06** [comment]
Thank for your advice @cjohnson57 ! I push a fix to resolve it.

File: tests/plando/boss_rush_dc_mq.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
   "shuffle_dungeon_entrances":               true,
   "open_forest":                             "open",
   "mq_dungeons":                             1,
+  "dungeon_shortcuts_choice":                "choice",
```
Comment (cjohnson57) [inline tests/plando/boss_rush_dc_mq.json:7]: Should be `choices`

File: SettingsList.py [hunk 5]
```diff
@@ -2430,22 +2430,16 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
-            'dodongos_cavern': 'Dodongo\'s Cavern',
-            'jabu_jabus_belly': 'Jabu Jabu\'s Belly',
-            'forest_temple': 'Forest Temple',
-            'fire_temple': 'Fire Temple',
-            'water_temple': 'Water Temple', # doesn't do anything, but added to prevent confusion
-            'shadow_temple': 'Shadow Temple',
-            'spirit_temple': 'Spirit Temple'
+            'off':       'Off',
+            'choice':    'Choice dungeons',
+            'all':	     'All dungeons',
```
Comment (cjohnson57) [inline SettingsList.py:2440]: A sneaky tab right here, should replace with spaces

File: SettingsList.py [hunk 6]
```diff
@@ -2430,22 +2430,16 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
-            'deku_tree': 'Deku Tree',
-            'dodongos_cavern': 'Dodongo\'s Cavern',
-            'jabu_jabus_belly': 'Jabu Jabu\'s Belly',
-            'forest_temple': 'Forest Temple',
-            'fire_temple': 'Fire Temple',
-            'water_temple': 'Water Temple', # doesn't do anything, but added to prevent confusion
-            'shadow_temple': 'Shadow Temple',
-            'spirit_temple': 'Spirit Temple'
+            'off':       'Off',
+            'choice':    'Choice dungeons',
```
Comment (cjohnson57) [inline SettingsList.py:2439]: Should be "Choose dungeons"

File: SettingsList.py [hunk 7]
```diff
@@ -2467,6 +2461,39 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             <b>Spirit Temple</b>: lobby elevator activated,
             shortcut silver blocks moved, central room
             platform lowered, and statue face melted
+
+            Choice: Select dungeons with shortcuts
```
Comment (cjohnson57) [inline SettingsList.py:2465]: Should also be choose

**Review by cjohnson57 — CHANGES_REQUESTED** [review]
Thanks for working to update it, just a few more things I failed to notice last time.

File: tests/plando/boss_rush_dc_mq.json (hunk 4 above)
Comment (RawZ06) [reply tests/plando/boss_rush_dc_mq.json:7]: What word should be in plural, the name of settings "dungeon_shortcuts_choices" or the name of property "choices" ?

File: SettingsList.py (hunk 5 above)
Comment (RawZ06) [reply SettingsList.py]: Fixed : https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/a0e2952eabb59bda65bd4ee3ca66f82250fcf08a

File: SettingsList.py (hunk 6 above)
Comment (RawZ06) [reply SettingsList.py]: Fixed : https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/a0e2952eabb59bda65bd4ee3ca66f82250fcf08a

File: SettingsList.py (hunk 7 above)
Comment (RawZ06) [reply SettingsList.py]: Fixed :https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/a0e2952eabb59bda65bd4ee3ca66f82250fcf08a

**Comment by RawZ06** [comment]
Thank to take the time to review my code. I did some fixes.

File: tests/plando/boss_rush_dc_mq.json (hunk 4 above)
Comment (cjohnson57) [reply tests/plando/boss_rush_dc_mq.json:7]: The name of the property, the same as it's `choices` in the other plandos

File: SettingsList.py [hunk 8]
```diff
@@ -2430,22 +2430,16 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
         },
     ),
-    Setting_Info(
-        name           = 'dungeon_shortcuts',
-        type           = list,
-        gui_text       = 'Dungeon Boss Shortcuts',
-        gui_type       = "MultipleSelect",
+    Combobox(
+        name           = 'dungeon_shortcuts_choice',
+        gui_text       = 'Dungeon Boss Shortcuts Choice',
```
Comment (cjohnson57) [inline SettingsList.py:2435]: To keep consistency with #1485, I'd rename the `gui_text` here to 'Dungeon Boss Shortcuts Mode', and...

File: SettingsList.py [hunk 9]
```diff
@@ -2467,6 +2461,39 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
+        disable={
+            'off': {'settings' : ['dungeon_shortcuts']},
+            'all': {'settings' : ['dungeon_shortcuts']},
+            'random': {'settings' : ['dungeon_shortcuts']},
+        },
+    ),
+    Setting_Info(
+        name           = 'dungeon_shortcuts',
+        type           = list,
+        gui_text       = 'Dungeon Boss Shortcuts List',
```
Comment (cjohnson57) [inline SettingsList.py:2479]: ... change `gui_text` here to just 'Dungeon Boss Shortcuts'

File: tests/plando/boss_rush_dc_mq.json (hunk 4 above)
Comment (RawZ06) [reply tests/plando/boss_rush_dc_mq.json:7]: Oh, i didn't understand, all plando with copy/paste I did a mistake, the option value is without 's' in SettingsList

Comment (RawZ06) [reply tests/plando/boss_rush_dc_mq.json:7]: Fix : https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1493/commits/ae9717c35a33e758bca64f3f9de721d194555bd9

**Comment by RawZ06** [comment]
Thank, I pushed a new commit to fix last things to modify.

**Review by cjohnson57 — APPROVED** [review]
Alright, seems good to me! Thanks for your quick fixes and putting up with all my nitpicks

**Pull request merged**